### DISCUSSION TITLE
How to query module event using ts-sdk?

### DISCUSSION THREAD

**Discussion by 0x-j** [aptos-labs/aptos-developer-discussions#6] [Questions]
I looked at ts sdk code and it's calling indexer endpoint to get events, so I went to [indexer's hasura](https://cloud.hasura.io/public/graphiql?endpoint=https://indexer.mainnet.aptoslabs.com/v1/graphql), and tried this query
```graphql
query MyQuery {
  events(offset: 10, limit: 10, where: {type: {_eq: ""}}) {
    type
    account_address
    creation_number
    data
    indexed_type
    sequence_number
    transaction_block_height
    event_index
    transaction_version
  }
}
```
Result looks like this, this is a pyth update price oracle event. But i can't tell if this is a module event or not from the response. Then I looked into [pyth contract](https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/aptos/contracts/sources/event.move#L29), and I think pyth still use event-handle event. 
```json
[
     {
        "type": "0x7e783b349d3e89cf5931af376ebeadbfab855b3fa239b7ada8f5a92fbea6b387::event::PriceFeedUpdate",
        "account_address": "0x7e783b349d3e89cf5931af376ebeadbfab855b3fa239b7ada8f5a92fbea6b387",
        "creation_number": 2,
        "indexed_type": "0x7e783b349d3e89cf5931af376ebeadbfab855b3fa239b7ada8f5a92fbea6b387::event::PriceFeedUpdate",
        "sequence_number": 93035016,
        "transaction_block_height": 112163800,
        "event_index": 0,
        "transaction_version": 325874498
      },
]
```
So how can we query module event from indexer (then I can update ts sdk if needed)?

**Comment by bowenyang007** [comment]
A couple of things: 
1. Please use `indexed_type` instead of type. Type field isn't indexed so it'll be less performant. 
2. Module events account_address, creation_number, and sequence number are all 0. [Here's an example](https://cloud.hasura.io/public/graphiql?endpoint=https%3A%2F%2Findexer.mainnet.aptoslabs.com%2Fv1%2Fgraphql&query=query+MyQuery+%7B%0A++events%28where%3A+%7Baccount_address%3A+%7B_eq%3A+%220x0000000000000000000000000000000000000000000000000000000000000000%22%7D%7D%29+%7B%0A++++account_address%0A++++creation_number%0A++++data%0A++++event_index%0A++++indexed_type%0A++++sequence_number%0A++++transaction_block_height%0A++++transaction_version%0A++++type%0A++%7D%0A%7D%0A)

**Comment by 0x-j** [comment]
Added an API to get module event in ts-sdk https://github.com/aptos-labs/aptos-ts-sdk/pull/304
